tests/e2e/scale/test_scale_pvc_pod_multi_project.py

```diff
@@ -0,0 +1,145 @@
+"""
+Testcase to Create 200 PVC+POD scale in one namespace with and without IO
+This will be Scale upto 1500 PVC+POD
+Scale PVC and POD Configuration
+---------------------------------------------------------
+Number of PVC+POD Creation: 1500
+PVC Type: All 4
+POD Type: Nginx basic configuration, no DC POD
+Cluster Capacity to be Utilized: 80%
+Network or IO saturation: 80%
+"""
+import logging
+import pytest
+import random
+
+from tests import helpers
+from ocs_ci.ocs import constants, cluster, machine, node
+from ocs_ci.ocs.resources import pod, pvc
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour, CephHealthException
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers
+
+log = logging.getLogger(__name__)
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames="start_io",
+    argvalues=[
+        pytest.param(
+            *[False], marks=pytest.mark.polarion_id("OCS-1357")
+        ),
+        pytest.param(
+            *[True], marks=pytest.mark.polarion_id("OCS-1357")
+        )
+    ]
+)
+class TestMultiProjectScalePVCPOD(E2ETest):
+    """
+    Scale the OCS cluster to reach 1500 PVC+POD
+    """
+    @pytest.fixture()
+    def setup_fixture(self, request):
+        def finalizer():
+            self.cleanup()
+
+        request.addfinalizer(finalizer)
+
+    def test_multi_project_scale_pvcs_pods(self, setup_fixture, start_io):
+        """
+        Test case to scale PVC+POD with multi projects and reach expected PVC count
+        :param setup_fixture:
+        :param start_io:
+        :return:
+        """
+
+        # Set PVC count based on number of OCS workers
+        ocs_nodes = node.get_typed_nodes(node_type='worker')
+        pvc_count = {3: 1500, 6: 3000, 9: 4500}
+        scale_count = pvc_count[len(ocs_nodes)]
+
+        # Other variables
+        rbd_sc_obj = helpers.default_storage_class(constants.CEPHBLOCKPOOL)
+        cephfs_sc_obj = helpers.default_storage_class(constants.CEPHFILESYSTEM)
+        pvc_count_each_itr = 5
+        size = f"{random.randrange(5, 105, 5)}Gi"
+        fio_size_param = '2G'
+        fio_rate_param = '16k'
+        self.namespace_list, all_pod_obj = ([] for i in range(2))
+
+        # Pre-requisite check for number of OSD's and app worker nodes.
+        helpers.add_required_osd_count(total_osd_nos=3)
+
+        # Create machineset for app worker nodes, which will create one app worker node
+        self.ms_name = machine.create_custom_machineset(instance_type='m5.4xlarge', zone='a')
```

Comment:
please don't hard code the instance type and zone.
Will it work on VMware?

Reply:
For now the machineset function works only for IPI, and existing machineset.yaml will work for only AWS, will update this name more generic.